### ISSUE TITLE
Pulling annotations from server

### ISSUE THREAD

**Issue by samuelpiltch** [nitaliano/react-native-mapbox-gl#565]
Was wondering if anyone knows how to pull JSON data from a server and create annotations from them? I have read the React Native API, and understand the way Annotations work, hoping they would work like markers in the web version. Currently, I have the following, which I know does not pull data and store it under initStops yet. However, when adding more than one object to the array, I receive an error that reads: _Exception thrown while executing UI block: - [RCTMapboxAnnotationannotationImageSource]: unrecognized selector sent to instance 0x7fc2115a9320_

Excerpt of Main.js:
```
render() {

        initStops = [
            {
                id: "A1",
                color: "blue",
                text: "A",
                coords: {latitude: 40, longitude: -120 }
            },
            {
                id: "A2",
                color: "blue",
                text: "A",
                coords: {latitude: 40, longitude: -120 }
            },

        ];

        var data;
        var xhr = new XMLHttpRequest();
        xhr.open('GET', 'http://localhost:5000', true);

        xhr.onload = function () {
            data = JSON.parse(xhr.responseText)
        };
        xhr.send();

        stopListArr = initStops.map(stopInfo => (
            <Stop
                key={stopInfo.id}
                id={stopInfo.id}
                line={stopInfo.text}
                lineColor={stopInfo.lineColor}
                coords={stopInfo.coords}
            />
        ))

        return (
            <View style={styles.container}>
                <MapView
                    style={styles.map}
                    styleURL={AccessCodes.mbstyle}
                    logoIsHidden={true}
                    compassIsHidden={true}

                    initialCenterCoordinate={{ latitude: 40.744683, longitude: -73.984853 }}
                    initialZoomLevel={14}
                    initialDirection={0}
                    rotateEnabled={true}
                    scrollEnabled={true}
                    zoomEnabled={true}

                    showsUserLocation={true}
                    userTrackingMode={Mapbox.userTrackingMode.follow}
                >
                    {stopListArr}
                </MapView>
            </View>
        );
    }
```

Excerpt from <Stop> component:
```
render() {
        return (
                <Annotation
                    id={this.props.id}
                    coordinate={this.props.coords}
                    style={{alignItems: 'center', justifyContent: 'center', position: 'absolute'}}
                >
                    <View style={styles.stop}>
                        <Text style={styles.text}>{this.props.line}</Text>
                    </View>
                </Annotation>
        );
    }
```

**Comment by rafaelcanovas** [comment]
Here's a working example.

`POI` meaning Point of Interest. API being any web service that you can make a request and get a response with the annotations data. `turf` and `debounce` are being used additionally but they're not necessary. 

The key point is to get annotations from the `state` object and every time the map is moved (`onRegionDidChange`) we get new annotations from the API and set again to the `state` object.

```js
import turf from 'turf';
import debounce from 'debounce';
import API from './app/modules/api';

export default class MapApplication extends Component {
    constructor () {
        super();

        this.state = {
            annotations: []
        };
    }

    loadPOI () {
        this.map.getBounds(bounds => {
            this.map.getCenterCoordinateZoomLevel(center => {
                let radius = turf.distance(
                    turf.point([bounds[3], bounds[2]]),
                    turf.point([bounds[1], bounds[0]])
                ) * 1500;

                API.pois.list({
                    center: center.latitude + ',' + center.longitude,
                    radius: radius
                }, response => {
                    let newAnnotations = response.data.results.map(p => {
                        return {
                            type: 'point',
                            id: p.uuid,
                            coordinates: [p.latitude, p.longitude]
                        };
                    });

                    this.setState({
                        annotations: newAnnotations
                    });
                });
            });
        });
    }

    render () {
        return (
            <View style={{flex: 1, backgroundColor: 'red'}}>
                <MapView style={{flex: 1}}
                         ref={map => { this.map = map; }}
                         initialZoomLevel={14}
                         annotations={this.state.annotations}
                         annotationsAreImmutable={true}
                         onRegionDidChange={debounce(this.loadPOI.bind(this), 200)}
                />
            </View>
        );
    }
}
```

If you have any questions please let me know.

**Comment by henrikra** [comment]
@samuelpiltch Can you close this if this has been solved?

**Comment by samuelpiltch** [comment]
Selector problem fixed with PR #555
